File: Leetcode/easy/maxProfit.java
class Solution {
    public int maxProfit(int[] prices) {
        // Time Complexity: O(N)
        // Space Complexity: O(1)

        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for (int idx = 0; idx < prices.length; idx++) {
            if (prices[idx] < minPrice) {
                minPrice = prices[idx];
            } else if (prices[idx] - minPrice > maxProfit) {
                maxProfit = prices[idx] - minPrice;
            }
        }

        return maxProfit;
    }
}

/*
 
Runtime
2 ms
Beats
87.15%
Memory
59 MB
Beats
82.49%

 */

File: Leetcode/easy/maxProfit.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # Time Complexity: O(N)
        # Space Complexity: O(1)
        
        minPrice = float('inf');
        maxProfit = 0

        for index in range (len(prices)):
            if prices[index] < minPrice:
                minPrice = prices[index]
            elif prices[index] - minPrice > maxProfit:
                maxProfit = prices[index] - minPrice

        return maxProfit

'''

Runtime
911 ms
Beats
99.74%
Memory
25 MB
Beats
87.39%

'''
